src/views/states/list/components/StateRow.tsx

```diff
@@ -2,37 +2,75 @@ import React, { FC } from 'react';
 import { NodeNetworkStateModelGroupVersionKind } from 'src/console-models';
 
 import { ResourceLink, RowProps, TableData } from '@openshift-console/dynamic-plugin-sdk';
-import { List, ListItem } from '@patternfly/react-core';
+import { List, ListItem, Title } from '@patternfly/react-core';
+import { ExpandableRowContent, Tbody, Td, Tr } from '@patternfly/react-table';
 import { V1beta1NodeNetworkState } from '@types';
+import { useNMStateTranslation } from '@utils/hooks/useNMStateTranslation';
 
-const StateRow: FC<RowProps<V1beta1NodeNetworkState>> = ({ obj, activeColumnIDs }) => {
-  const typesCount = obj?.status?.currentState?.interfaces.reduce((acc, iface) => {
+import InterfacesTable from './InterfacesTable';
+
+const StateRow: FC<RowProps<V1beta1NodeNetworkState, { rowIndex: number }>> = ({
+  obj,
+  activeColumnIDs,
+  rowData: { rowIndex },
+}) => {
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const { t } = useNMStateTranslation();
+  const interfaces = obj?.status?.currentState?.interfaces;
+
+  const interfacesByType = interfaces?.reduce((acc, iface) => {
     if (acc[iface.type]) {
-      acc[iface.type] += 1;
+      acc[iface.type].push(iface);
     } else {
-      acc[iface.type] = 1;
+      acc[iface.type] = [iface];
     }
     return acc;
   }, {} as { [key: string]: number });
 
   return (
-    <>
-      <TableData id="name" activeColumnIDs={activeColumnIDs} className="pf-m-width-30">
-        <ResourceLink
-          groupVersionKind={NodeNetworkStateModelGroupVersionKind}
-          name={obj.metadata.name}
+    <Tbody key={obj.metadata.name} isExpanded={isExpanded} role="rowgroup">
+      <Tr>
+        <Td
+          expand={{
+            rowIndex,
+            isExpanded,
+            onToggle: (event, rowIndex, isOpen) => setIsExpanded(isOpen),
+            expandId: 'expand-interfaces-list',
+          }}
         />
-      </TableData>
-      <TableData id="network-interface" activeColumnIDs={activeColumnIDs} className="pf-m-width-50">
-        <List isPlain>
-          {Object.keys(typesCount)?.map((interfaceType) => (
-            <ListItem key={interfaceType}>
-              {interfaceType} ({typesCount[interfaceType]})
-            </ListItem>
-          ))}
-        </List>
-      </TableData>
-    </>
+        <TableData id="name" activeColumnIDs={activeColumnIDs}>
+          <ResourceLink
+            groupVersionKind={NodeNetworkStateModelGroupVersionKind}
+            name={obj.metadata.name}
```

Comment:
I recommend to add also `namespace` prop to `ResourceLink`. 
Same as I did just now for example here:
https://github.com/kubevirt-ui/kubevirt-plugin/pull/1324

Reply:
Here we are using cluster level resources. They don't have a namespace. This is also the reason why we have no `namespace` select at the top of the page.

Reply:
Ah, okay, good point, thanks for the explanation, Ugo! :)